nifi-commons/nifi-hadoop-utils/src/main/java/org/apache/nifi/hadoop/SecurityUtil.java

```diff
@@ -0,0 +1,113 @@
+     * UserGroupInformation.
+     *
+     * @param config the configuration instance
+     * @param principal the principal to authenticate as
+     * @param keyTab the keytab to authenticate with
+     *
+     * @return the UGI for the given principal
+     *
+     * @throws IOException if login failed
+     */
+    public static synchronized UserGroupInformation loginKerberos(final Configuration config, final String principal, final String keyTab)
+            throws IOException {
+        Validate.notNull(config);
+        Validate.notNull(principal);
+        Validate.notNull(keyTab);
+
+        UserGroupInformation.setConfiguration(config);
+        return UserGroupInformation.loginUserFromKeytabAndReturnUGI(principal.trim(), keyTab.trim());
+    }
+
+    /**
+     * Initializes UserGroupInformation with the given Configuration and returns UserGroupInformation.getLoginUser().
+     * All logins should happen through this class to ensure other threads are not concurrently modifying
+     * UserGroupInformation.
+     *
+     * @param config the configuration instance
+     *
+     * @return the UGI for the given principal
+     *
+     * @throws IOException if login failed
+     */
+    public static synchronized UserGroupInformation loginSimple(final Configuration config) throws IOException {
+        Validate.notNull(config);
+        UserGroupInformation.setConfiguration(config);
+        return UserGroupInformation.getLoginUser();
+    }
+
+    /**
+     * Initializes UserGroupInformation with the given Configuration and returns UserGroupInformation.isSecurityEnabled().
+     *
+     * All checks for isSecurityEnabled() should happen through this method.
+     *
+     * @param config the given configuration
+     *
+     * @return true if kerberos is enabled on the given configuration, false otherwise
+     *
+     */
+    public static synchronized boolean isSecurityEnabled(final Configuration config) {
+        Validate.notNull(config);
+        return "kerberos".equalsIgnoreCase(config.get("hadoop.security.authentication"));
+    }
+
+    /**
+     * Start a thread that periodically attempts to renew the current Kerberos user's ticket.
+     *
+     * Callers of this method should store the reference to the KerberosTicketRenewer and call stop() to stop the thread.
+     *
+     * @param clazz
+     *          The class that this renewal is for (i.e. PutHDFS, etc)
+     * @param ugi
+     *          The current Kerberos user.
+     * @param renewalPeriod
+     *          The amount of time between attempting renewals.
+     * @param logger
+     *          The logger to use with in the renewer
+     *
+     * @return the KerberosTicketRenewer Runnable
+     */
+    public static KerberosTicketRenewer startTicketRenewalThread(final Class clazz, final UserGroupInformation ugi, final long renewalPeriod, final ComponentLog logger) {
```

Comment:
Given that the class is used only for the thread name, I think it would make more sense to just pass in a String for the name, no?